compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -243,16 +245,34 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
                 ty::PredicateKind::Clause(ty::Clause::RegionOutlives(predicate)) => {
                     self.compute_region_outlives_goal(Goal { param_env, predicate })
                 }
+                ty::PredicateKind::Subtype(predicate) => {
+                    self.compute_subtype_goal(Goal { param_env, predicate })
+                }
+                ty::PredicateKind::Coerce(predicate) => self.compute_subtype_goal(Goal {
+                    param_env,
+                    predicate: SubtypePredicate {
+                        a_is_expected: true,
+                        a: predicate.a,
+                        b: predicate.b,
+                    },
+                }),
```

Comment:
can you still move that into its own function. Coerce predicates surely feel nonsensical but it also really just doesn't matter for the rest of the solver so I want to keep the complexity of `compute_goal` smaller.